### PR TITLE
Trra 158/cap remaning

### PR DESCRIPTION
**Pull request by aprigge**
-Adds box on employee page to indicate remaining professional development balances

-Adds `profdev_spent` and `profdev_days_away` to `employee_total_dictionary`
-Adds `profdev_remaining` and `profdev_days_remaining` to `employee_total_report`. If days remaining is less than 0, `profdev_days_remaining` is 0.

(Feel free to suggest any styling changes too)

### CODE REVIEW

File: terra/reports.py [hunk 1]
```diff
@@ -704,22 +736,46 @@ def get_individual_data_employee(employee_ids, start_date=None, end_date=None):
...
+        )
     )
     return rows
 
 
 def employee_total_report(employee_ids, start_date, end_date):
     employee_totals = {}
     rows = get_individual_data_employee(employee_ids, start_date, end_date)
+    profdev_spending_cap = 3500
+    profdev_days_cap = 15
```
Comment (akohler) [inline terra/reports.py]: @aprigge I think these "magic numbers" are spreading.  For example, these also appear in `terra/templatetags/terra_extras.py`.  Can they be pulled into a central location and defined as constants, then referenced as needed wherever relevant?  Maybe they can fit into a model; possibly a context, maybe something else?

If the spending cap goes up to $3800 next year, for example, we don't want to hunt down and change every use of 3500 in the project.

Which raises the question: as these caps change over time, will previous FY reports be affected?  I don't think so, other than highlighting / color coding, but something to consider.

If there's no immediate obvious best way to handle this centralization, I'm OK merging this now, as long as a ticket's created to address the resulting technical debt.

Thanks --Andy

Comment (aprigge) [reply terra/reports.py]: @akohler, good point. Let me centralize the caps before merging. But yes, in general we haven't figured out what to do when/if caps change. Thanks!

**Comment by aprigge** [comment]
@akohler, I've updated this branch so that the caps are centralized in the utils file.

File: terra/templatetags/terra_extras.py [hunk 2]
```diff
@@ -47,11 +53,11 @@ def days_cap(value):
...
         value = 0
         return value
-    if value >= 15:
+    # Over or at professional development cap
+    if value >= profdev_days_cap:
         return '<span class="alert-danger">{}</span>'.format(value)
-        # Over or at professional development cap
-    elif value >= 12 and value < 15:
+    # Within 20% of professional development cap
+    elif value >= profdev_days_warning and value < profdev_days_cap:
```
Comment (akohler) [inline terra/templatetags/terra_extras.py]: Thanks for catching and changing the warning amounts too

**Pull request merged**